packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -952,6 +958,21 @@ async function generateArmParameterJson(ctx: CoreHookContext) {
     parameterObj[ArmParameters.botWebAppSku] =
       envConfig[ResourcePlugins.Bot]?.[EnvConfigName.SkuName];
   }
+
+  // Apim
+  if (envConfig[ResourcePlugins.Apim]?.[EnvConfigName.ServiceName]) {
+    parameterObj[ArmParameters.ApimServiceName] =
+      envConfig[ResourcePlugins.Apim]?.[EnvConfigName.ServiceName];
+  }
+  if (envConfig[ResourcePlugins.Apim]?.[EnvConfigName.ProductId]) {
+    parameterObj[ArmParameters.ApimProductName] =
+      envConfig[ResourcePlugins.Apim]?.[EnvConfigName.ProductId];
+  }
+  if (envConfig[ResourcePlugins.Apim]?.[EnvConfigName.OAuthServerId]) {
+    parameterObj[ArmParameters.ApimOauthServerName] =
+      envConfig[ResourcePlugins.Apim]?.[EnvConfigName.OAuthServerId];
+  }
+
```

Comment:
do we need to convert the service name (as well as other names) to apiServiceResourceId?

Reply:
fixed in latest commit.